core/crypto/vrf/vrf_provider_impl.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/vrf/vrf_provider_impl.hpp"
+
+#include <gsl/span>
+
+namespace kagome::crypto {
+
+  VRFProviderImpl::VRFProviderImpl(std::shared_ptr<CSPRNG> generator)
+      : generator_{std::move(generator)} {}
+
+  SR25519Keypair VRFProviderImpl::generateKeypair() const {
+    auto seed = generator_->randomBytes(SR25519_SEED_SIZE);
+
+    std::vector<uint8_t> kp(SR25519_KEYPAIR_SIZE, 0);
+    sr25519_keypair_from_seed(kp.data(), seed.data());
+
+    return SR25519Keypair{kp};
+  }
+
+  namespace {
+    std::array<uint8_t, 32> uint256_t_to_bytes(
+        const boost::multiprecision::uint256_t &i) {
+      auto count = i.backend().size();
+      auto tsize = sizeof(boost::multiprecision::limb_type);
+      auto copy_count = count * tsize;
+
+      std::array<uint8_t, 32> output{};
+      memcpy(output.data(), i.backend().limbs(), copy_count);
+      return output;
+    }
+
+    boost::multiprecision::uint256_t bytes_to_uint256_t(
+        gsl::span<uint8_t, 32> bytes) {
+      boost::multiprecision::uint256_t i;
+      auto size = bytes.size();
+      i.backend().resize(size, size);
+      memcpy(i.backend().limbs(), bytes.data(), bytes.size());
+      i.backend().normalize();
+      return i;
+    }
+  }  // namespace
```

Comment:
Actually, I need those functions as well in my current code. Please, move them to some kind of "crypto/util.hpp"